escheduler-server/src/main/java/cn/escheduler/server/worker/task/http/HttpTask.java

```diff
@@ -0,0 +1,310 @@
+        return webPage;
+    }
+
+    protected int getStatusCode(CloseableHttpResponse httpResponse) {
+        int status = httpResponse.getStatusLine().getStatusCode();
+        return status;
+    }
+
+    protected int validResponse(String body, String statusCode) throws Exception {
+        int exitStatusCode = 0;
+        switch (httpParameters.getHttpCheckCondition()) {
+            case BODY_CONTAINS:
+                if (StringUtils.isEmpty(body) || !body.contains(httpParameters.getCondition())) {
+                    appendMessage(httpParameters.getUrl() + " doesn contain "
+                            + XmlEscapers.xmlContentEscaper().escape(httpParameters.getCondition()));
+                    exitStatusCode = -1;
+                }
+                break;
+            case BODY_NOT_CONTAINS:
+                if (StringUtils.isEmpty(body) || body.contains(httpParameters.getCondition())) {
+                    appendMessage(httpParameters.getUrl() + " contains "
+                            + XmlEscapers.xmlContentEscaper().escape(httpParameters.getCondition()));
+                    exitStatusCode = -1;
+                }
+                break;
+            case STATUS_CODE_CUSTOM:
+                if (!statusCode.equals(httpParameters.getCondition())) {
+                    appendMessage(httpParameters.getUrl() + " statuscode: " + statusCode + ", Must be: " + httpParameters.getCondition());
+                    exitStatusCode = -1;
+                }
+                break;
+            default:
+                if (!"200".equals(statusCode)) {
+                    appendMessage(httpParameters.getUrl() + " statuscode: " + statusCode + ", Must be: 200");
+                    exitStatusCode = -1;
+                }
+                break;
+        }
+        return exitStatusCode;
+    }
+
+    public String getOutput() {
+        return output;
+    }
+
+    protected void appendMessage(String message) {
+        if (output == null) {
+            output = "";
+        }
+        if (message != null && !message.trim().isEmpty()) {
+            output += message;
+        }
+    }
+
+    protected void addRequestParams(RequestBuilder builder,Map<String, Property> paramsMap) {
+        Iterator<Map.Entry<String, Property>> iter = paramsMap.entrySet().iterator();
+        while (iter.hasNext()){
+            Map.Entry<String, Property> en = iter.next();
+            Property property = en.getValue();
+            if (property.getValue() != null && property.getValue().length() > 0){
+                if (property.getDirect().equals(Direct.IN)){
+                    builder.addParameter(property.getProp(), property.getValue());
+                }else{
+                    if(CONTENT_TYPE.equals(property.getProp()) && property.getValue().contains(APPLICATION_JSON)){
+                        StringEntity postingString = null;
+                        try {
+                            postingString = new StringEntity("{}");
```

Comment:
You can just specify the Charset, to avoid the UnsupportedEncodingException, like:    StringEntity a = new StringEntity("", Charsets.UTF_8);